src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -258,34 +262,41 @@ bool Compiler::optRedundantBranch(BasicBlock* const block)
         return false;
     }
 
-    // Bail out if tree is has certain side effects
+    // Be conservative if there is an exception effect and we're in an EH region
```

Comment:
Note to self: not sure why we don't check this condition earlier. No point in searching for dominating compares if we're just going to bail out here like this.